internal/dlasqtest/dlasq6.f

```diff
@@ -0,0 +1,258 @@
+*> \verbatim
+*>
+*> DLASQ6 computes one dqd (shift equal to zero) transform in
+*> ping-pong form, with protection against underflow and overflow.
+*> \endverbatim
+*
+*  Arguments:
+*  ==========
+*
+*> \param[in] I0
+*> \verbatim
+*>          I0 is INTEGER
+*>        First index.
+*> \endverbatim
+*>
+*> \param[in] N0
+*> \verbatim
+*>          N0 is INTEGER
+*>        Last index.
+*> \endverbatim
+*>
+*> \param[in] Z
+*> \verbatim
+*>          Z is DOUBLE PRECISION array, dimension ( 4*N )
+*>        Z holds the qd array. EMIN is stored in Z(4*N0) to avoid
+*>        an extra argument.
+*> \endverbatim
+*>
+*> \param[in] PP
+*> \verbatim
+*>          PP is INTEGER
+*>        PP=0 for ping, PP=1 for pong.
+*> \endverbatim
+*>
+*> \param[out] DMIN
+*> \verbatim
+*>          DMIN is DOUBLE PRECISION
+*>        Minimum value of d.
+*> \endverbatim
+*>
+*> \param[out] DMIN1
+*> \verbatim
+*>          DMIN1 is DOUBLE PRECISION
+*>        Minimum value of d, excluding D( N0 ).
+*> \endverbatim
+*>
+*> \param[out] DMIN2
+*> \verbatim
+*>          DMIN2 is DOUBLE PRECISION
+*>        Minimum value of d, excluding D( N0 ) and D( N0-1 ).
+*> \endverbatim
+*>
+*> \param[out] DN
+*> \verbatim
+*>          DN is DOUBLE PRECISION
+*>        d(N0), the last value of d.
+*> \endverbatim
+*>
+*> \param[out] DNM1
+*> \verbatim
+*>          DNM1 is DOUBLE PRECISION
+*>        d(N0-1).
+*> \endverbatim
+*>
+*> \param[out] DNM2
+*> \verbatim
+*>          DNM2 is DOUBLE PRECISION
+*>        d(N0-2).
+*> \endverbatim
+*
+*  Authors:
+*  ========
+*
+*> \author Univ. of Tennessee 
+*> \author Univ. of California Berkeley 
+*> \author Univ. of Colorado Denver 
+*> \author NAG Ltd. 
+*
+*> \date September 2012
+*
+*> \ingroup auxOTHERcomputational
+*
+*  =====================================================================
+      SUBROUTINE DLASQ6( I0, N0, Z, PP, DMIN, DMIN1, DMIN2, DN,
+     $                   DNM1, DNM2 )
+*
+*  -- LAPACK computational routine (version 3.4.2) --
+*  -- LAPACK is a software package provided by Univ. of Tennessee,    --
+*  -- Univ. of California Berkeley, Univ. of Colorado Denver and NAG Ltd..--
+*     September 2012
+*
+*     .. Scalar Arguments ..
+      INTEGER            I0, N0, PP
+      DOUBLE PRECISION   DMIN, DMIN1, DMIN2, DN, DNM1, DNM2
+*     ..
+*     .. Array Arguments ..
+      DOUBLE PRECISION   Z( * )
+*     ..
+*
+*  =====================================================================
+*
+*     .. Parameter ..
+      DOUBLE PRECISION   ZERO
+      PARAMETER          ( ZERO = 0.0D0 )
+*     ..
+*     .. Local Scalars ..
+      INTEGER            J4, J4P2
+      DOUBLE PRECISION   D, EMIN, SAFMIN, TEMP
+*     ..
+*     .. External Function ..
+      DOUBLE PRECISION   DLAMCH
+      EXTERNAL           DLAMCH
+*     ..
+*     .. Intrinsic Functions ..
+      INTRINSIC          MIN
+*     ..
+*     .. Executable Statements ..
+*
+      IF( ( N0-I0-1 ).LE.0 )
+     $   RETURN
+*
+
+      print *, "In dlasq6"
+      STOP
```

Comment:
I'm not sure. None of the random test cases I generated ever reached dlasq6. My error rate was low in translation, but not non-zero (and sometimes in tricky ways). Part of me wants a trigger in dlasq6.go so when a case is found we can debug it.